auth/src/main/java/com/firebase/ui/auth/ui/email/RecoverPasswordActivity.java

```diff
@@ -54,51 +56,57 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.fui_forgot_password_layout);
 
-        mEmailFieldValidator =
-                new EmailFieldValidator((TextInputLayout) findViewById(R.id.email_layout));
+        mHandler = ViewModelProviders.of(this).get(RecoverPasswordHandler.class);
+        mHandler.init(getFlowHolder());
+        mHandler.getPasswordResetListener().observe(this, new Observer<Task<String>>() {
```

Comment:
Hmm this seems a little strange to me.  We call a `getListener` method and then `observe` a listener?  

I think what we want is something like an `Observable<PasswordResetState>` and the state can include properties like `boolean success  and `String errorMessage`.  Also avoids passing down the `Task` and lets us encapsulate the error `instanceof` logic higher up.

Reply:
Yeah, this was poorly named.

I really like your idea of passing around a state object which includes everything needed to update the UI—I was always a little annoyed at having to observe two `LiveData`s. I'll play around with that in just a bit, thanks! 😄

Reply:
Ooooo, I'm super excited! Dunno why I didn't think of doing that from the start, but your idea makes things _waaaaaaaay_ nicer IMO. It also replaces the `DialogFragment` which is really cool and interesting. Whad'ya think? 😄